core/src/main/clojure/xtdb/metadata.clj

```diff
@@ -351,6 +354,21 @@
         (when (.getObject bloom-rdr bloom-vec-idx)
           (bloom/bloom->bitmap bloom-rdr bloom-vec-idx)))))
 
+  (temporalBounds[_ page-idx]
+    ;; TODO when badly distributed there are pages with no data
+    (if-let [temporal-col-types-rdr (-> (.structKeyReader metadata-leaf-rdr "columns")
```

Comment:
Try to push reader creation to constructor.

Reply:
Tried to push this into the constructor, but running into issues with a nil `temporal-col-types-rdr`. Some issue when data is badly distributed. I thought this is fixed, but there still seem to be cases when the metadata doesn't exist.

Reply:
If a trie is empty then these readers don't exist.